File: ReverseInteger/reverse_integer.go
package main

import "math"

func reverse(x int) int {

	var neg bool
	if x < 0 {
		neg = true
		x = x * -1
	}

	revX := getReversed(x)
	if neg == true {
		return revX * -1
	}
	return revX
}

func getReversed(x int) int {
	var rev, lastDigit int
	for x > 0 {
		lastDigit = x % 10
		x = x / 10
		rev = rev*10 + lastDigit
	}
	if rev > (math.MaxInt32) {
		return 0
	}
	return rev
}

File: ReverseInteger/reverse_integer.py
class Solution:
    def reverse(self, x: int) -> int:
        neg = False
        if x < 0:
            neg = True
            x = abs(x)

        revX = getReversed(x)

        if neg:
            return -revX
        return revX

def getReversed(x: int) -> int:
        rev = 0
        while x > 0:
            lastDigit = x % 10
            x //= 10
            rev = rev * 10 + lastDigit

        if rev > 2**31 - 1:  
            return 0
        return rev
